### DISCUSSION TITLE
How to customize distance calc functions?

### DISCUSSION THREAD

**Discussion by pwli0755** [milvus-io/milvus#18429] [Q&A and General discussion]
How to customize distance calc functions?

how to build the milvus-standalone docker image using the customized code?

**Comment by yhmo** [comment]
It is not allowed to customize distance calc function in milvus.
So far milvus supports these distance calc functions(we say metric methods):
- For float vectors: L2(Euclid Distance), IP(Inner product)
- For binary vectors: HAMMIN, JACCARD, TANIMOTO, SUBSTRUCTURE, SUPERSTRUCTURE
